packages/x-charts/src/internals/computeAxisValue.ts

```diff
@@ -198,17 +198,19 @@ export function computeAxisValue({
 
     const scaleType = axis.scaleType ?? ('linear' as const);
 
+    const domainLimit = axis.domainLimit ?? 'nice';
+
     const axisExtremums =
-      typeof axis.domainLimit === 'function'
-        ? axis.domainLimit([minData, maxData])
+      typeof domainLimit === 'function'
+        ? domainLimit([minData, maxData])
         : [axis.min ?? minData, axis.max ?? maxData];
     const rawTickNumber = getTickNumber({ ...axis, range, domain: axisExtremums });
     const tickNumber = rawTickNumber / ((zoomRange[1] - zoomRange[0]) / 100);
 
     const zoomedRange = zoomScaleRange(range, zoomRange);
 
     const scale = getScale(scaleType, axisExtremums, zoomedRange);
-    const finalScale = axis.domainLimit === 'strict' ? scale : scale.nice(rawTickNumber);
+    const finalScale = domainLimit === 'nice' ? scale.nice(rawTickNumber) : scale;
```

Comment:
This fix line was overriding the function case, because when `axis.domainLimit` is a function, it's not equal to `'strinct'` and so the `.nice()` was applied (overriding the user choice)